crates/pathfinder/src/state/sync/pending.rs

```diff
@@ -78,17 +78,14 @@ pub async fn poll_pending<S: GatewayApi + Clone + Send + 'static>(
                     prev_block = Some(block.clone());
                     tracing::trace!("Pending block data changed");
 
```

Comment:
This still looks fishy -- `prev_block` is updated even if downloading classes fails later.

Reply:
Hmm right we would skip the next iteration as well then if there were no updates. Good catch.

Reply:
I think I've solved this by simplifying the logic a bit.